Remove Automatic Snapshot Deployment to GCR for older versions

matrei opened the pull request:
Only the latest release branch should automatically deploy snapshots to https://start.grails.org.

matrei committed:
ci: Remove automatic snapshot publishing

Only the latest release branch should automatically publish snapshots as it updates https://start.grails.org.

jdaugherty commented on .github/workflows/snapshot.yml:
```diff
@@ -1,11 +1,9 @@
 name: Snapshot
 on:
-  push:
```
Shouldn't the CLI still be published as a snapshot?  It's only the google deploy we want to avoid ... right?

jdaugherty committed:
[Backport] Do not default the java version from the grails cli version (#434)

Decouples the graalvm version used for compiling the grails-cli from the
java version used in the projects it creates.

matrei committed:
ci: feedback - only prevent deploy to GCR

Change to only prevent the deployment to Google Cloud Run.

We still want to the snapshot workflow to run as it builds, tests, publishes to sonatype ossrh, and uploads the built artifacts in the workflow summary.

matrei committed:
ci: Update snapshot.yml to latest standards

matrei committed:
build: Update develocity gradle plugins

matrei committed:
fix: Compatible with Java 17

matrei committed:
build: add `org.graalvm.buildtools.native` gradle plugin

matrei committed:
chore: fix checkstyle errors

jdaugherty commented on .github/workflows/snapshot.yml:
```diff
@@ -9,287 +9,330 @@ on:
-    name: Deploy To Google Cloud Run
+    # Snapshots should only be deployed from the latest release branch
+    # as it updates the snapshot version on https://start.grails.org.
+    if: false # github.event_name != 'pull_request'
```
Is there a reason we're leaving the deploy step here if we're going to hard code it to false?  From what I can tell you moved it to the release.yml step, how do we ensure the latest 7.0.0 build gets built if it's not released yet?

matrei replied:
> Is there a reason we're leaving the deploy step here if we're going to hard code it to false?

Well, I could remove it instead but I thought it was self-documenting having it there. And if you, for some reason, did need to deploy, you and just flip the switch.

Would you rather have me remove it?

> From what I can tell you moved it to the release.yml step

 I don't think I moved it to `release.yml`. It was already there.

> how do we ensure the latest 7.0.0 build gets built if it's not released yet?

I'm not sure I understand the question. This PR is for the 6.0.x branch. I am planning to merge it up to 6.1.x and 6.2.x making changes as necessary. The 7.0.x branch will not get any changes to GCR publishing at the moment.

jdaugherty replied:
I understand now, lets leave it?
